osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/datadecoder/BasicDlmsDataDecoderTest.java

```diff
@@ -0,0 +1,101 @@
+// SPDX-FileCopyrightText: Copyright Contributors to the GXF project
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package org.opensmartgridplatform.adapter.protocol.dlms.domain.datadecoder;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.openmuc.jdlms.datatypes.CosemDate;
+import org.openmuc.jdlms.datatypes.CosemDateTime;
+import org.openmuc.jdlms.datatypes.CosemDateTime.ClockStatus;
+import org.openmuc.jdlms.datatypes.CosemTime;
+import org.openmuc.jdlms.datatypes.DataObject;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.commands.utils.DlmsHelper;
+import org.opensmartgridplatform.adapter.protocol.dlms.exceptions.ProtocolAdapterException;
+
+class BasicDlmsDataDecoderTest {
+  private final DlmsHelper dlmsHelper = new DlmsHelper();
+  private final BasicDlmsDataDecoder decoder = new BasicDlmsDataDecoder(this.dlmsHelper);
+
+  @ParameterizedTest
+  @EnumSource(
```

Comment:
This is not a parameterized test. Every DataObjerct.Type in the enumSource has it's own test. Test might as well be separated into test fo each DataObject.Type

Reply:
Changed to separated tests